fhir-model/src/main/java/com/ibm/fhir/model/util/XMLSupport.java

```diff
@@ -197,15 +209,32 @@ private static XMLInputFactory createXMLInputFactory() {
 
     private static XMLOutputFactory createXMLOutputFactory() {
         try {
-            return (XMLOutputFactory) Class.forName(XML_OUTPUT_FACTORY_IMPL).getConstructor().newInstance();
+            boolean isSet = System.getProperty(PROP_XML_OUTPUT_FACTORY) != null;
+            if (!isSet) {
+                System.setProperty(PROP_XML_OUTPUT_FACTORY, XML_OUTPUT_FACTORY_IMPL);
+            }
+            XMLOutputFactory factory = XMLOutputFactory.newFactory();
```

Comment:
Just a thought here (although perhaps this is overly conservative)...if `!isSet`, then do a little loop and check the result of getProperty matches the  `XML_OUTPUT_FACTORY_IMPL` value after the newFactory call. As discussed, the race condition here is pretty theoretical, so this may be overkill.

Reply:
looping seems overkill to me, but it wouldn't hurt to check the value after we instantiate the factory.  the question is what would we do if it doesn't match...throw?

Reply:
I'm thinking maybe just a WARNING message for that case, but the more I think about it, the more I feel it shouldn't be needed.  I think we should merge as-is, but do let me know if you want the added check.

Reply:
I'm OK as-is...just throwing out a possible option for completeness.